context.go

```diff
@@ -32,290 +65,45 @@ func (c *Context) FindVariable(e Expression) (Expression, error) {
 		return nil, fmt.Errorf("DUNNO WHAT TO DO WITH %v OMG", e)
 	}
 
-	if value, ok := c.vars[key]; ok {
+	scope, err := c.scopes.curr()
+	if err != nil {
+		return nil, err
+	}
+
+	if value, ok := scope[key]; ok {
```

Comment:
Erm, sorry I jumped the gun.  There is a difference in how it's treated, but it's probably not the best way to handle it. Incoming commit to make Get work the same way and propagate the error up, which will make it so the caller can handle it (and make judgement calls on whether it's strict or lax handling).